pkg/operator/controller/machineconfig/crio.go

```diff
@@ -0,0 +1,201 @@
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+const (
+	// CrioProfilingConfigName is the name CRI-O MachineConfig CR
+	CrioProfilingConfigName = "99-crio-profiling"
+
+	// CrioUnixSocketConfFile is the name of the CRI-O config file
+	CrioUnixSocketConfFile = "10-mco-profile-unix-socket.conf"
+
+	// CriounixSocketConfData contains the configuration required
+	// for enabling CRI-O profiling
+	CriounixSocketConfData = `[Service]
+Environment="ENABLE_PROFILE_UNIX_SOCKET=true"`
+)
+
+// ensureCrioProfConfigExists checks if CRI-O MachineConfig CR for
+// enabling profiling exists, if not creates the resource
+func (r *MachineconfigReconciler) ensureCrioProfConfigExists(ctx context.Context) (*mcv1.MachineConfig, bool, error) {
+	namespace := types.NamespacedName{Name: CrioProfilingConfigName}
+	criomc, exists, err := r.fetchCrioProfileConfig(ctx, namespace)
+	if err != nil {
+		return nil, false, err
+	}
+	if !exists {
+		if err := r.createCrioProfileConfig(ctx); err != nil {
+			return nil, false, err
+		}
+
+		criomc, found, err := r.fetchCrioProfileConfig(ctx, namespace)
+		if err != nil || !found {
+			return nil, false, fmt.Errorf("failed to fetch just created crio config: %v", err)
+		}
+
+		return criomc, true, nil
+	}
+	return criomc, false, nil
+}
+
+// ensureCrioProfConfigNotExists checks if CRI-O MachineConfig CR for
+// enabling profiling exists, if exists delete the resource
+func (r *MachineconfigReconciler) ensureCrioProfConfigNotExists(ctx context.Context) (bool, error) {
+	namespace := types.NamespacedName{Name: CrioProfilingConfigName}
+	criomc, exists, err := r.fetchCrioProfileConfig(ctx, namespace)
+	if err != nil {
+		return false, err
+	}
+	if exists {
+		if err := r.deleteCrioProfileConfig(ctx, criomc); err != nil {
+			return false, err
+		}
+
+		_, exists, err = r.fetchCrioProfileConfig(ctx, namespace)
+		if err != nil || exists {
+			return false, fmt.Errorf("failed to delete crio profiling config: %v", err)
+		}
+
+		return true, nil
+	}
+	return false, nil
+}
+
+// fetchCrioProfileConfig is for fetching the CRI-O MC CR created
+// by this controller for enabling profiling
+func (r *MachineconfigReconciler) fetchCrioProfileConfig(ctx context.Context, namespace types.NamespacedName) (*mcv1.MachineConfig, bool, error) {
+	criomc := &mcv1.MachineConfig{}
+
+	if err := r.Get(ctx, namespace, criomc); err != nil {
+		if errors.IsNotFound(err) {
+			return nil, false, nil
+		}
+		return nil, false, err
+	}
+	return criomc, true, nil
+}
+
+// createCrioProfileConfig is for creating CRI-O MC CR
+func (r *MachineconfigReconciler) createCrioProfileConfig(ctx context.Context) error {
+	criomc, err := r.getCrioConfig()
+	if err != nil {
+		return err
+	}
+
+	if err := r.Create(ctx, criomc); err != nil {
+		return fmt.Errorf("failed to create crio profiling config %s: %w", criomc.Name, err)
+	}
+
+	// TODO: cluster-scoped resource must not have a namespace-scoped owner,
```

Comment:
I think @saschagrunert said it should be cluster scoped

https://github.com/openshift/node-observability-operator/pull/16#discussion_r833178413

Reply:
NodeObservabilityMachineConfig is now made as cluster-scoped resource.